x-pack/elastic-agent/pkg/composable/controller.go

```diff
@@ -107,6 +114,10 @@ func (c *controller) Run(ctx context.Context, cb VarsCallback) error {
 			cancel()
 			return errors.New(err, fmt.Sprintf("failed to run provider '%s'", name), errors.TypeConfig, errors.M("provider", name))
 		}
+		// manually register providers that want to pass to the Vars as "fetch" providers
```

Comment:
This should be done using that same register pattern.

Reply:
Changed it based on https://github.com/elastic/beats/pull/24789#discussion_r603410573